lib/internal/source_map/source_map_cache.js

```diff
@@ -89,7 +91,7 @@ function maybeCacheSourceMap(filename, content, cjsModuleInstance) {
     } else {
       // If there is no cjsModuleInstance assume we are in a
       // "modules/esm" context.
-      esmSourceMapCache.set(filename, {
+      MapPrototypeSet(esmSourceMapCache, filename, {
```

Comment:
No need if you use a `SafeMap`
```suggestion
      esmSourceMapCache.set(filename, {
```